src/Core/PostgreSQLProtocol.h

```diff
@@ -578,6 +582,140 @@ class Query : FrontMessage
     }
 };
 
+class ParseQuery : FrontMessage
+{
+public:
+    String function_name;
+    String sql_query;
+    Int16 num_params;
+
+    void deserialize(ReadBuffer & in) override
+    {
+        Int32 sz;
+        readBinaryBigEndian(sz, in);
+        readNullTerminated(function_name, in);
+        readNullTerminated(sql_query, in);
+        readBinaryBigEndian(num_params, in);
+        Int32 oid_param;
+        for (int i = 0; i < num_params; ++i)
+            readBinaryBigEndian(oid_param, in);
+    }
+
+    MessageType getMessageType() const override
+    {
+        return MessageType::PARSE;
+    }
+};
+
+class ParseQueryComplete : BackendMessage
+{
+public:
+    ParseQueryComplete() = default;
+
+    void serialize(WriteBuffer & out) const override
+    {
+        out.write('1');
+        writeBinaryBigEndian(size(), out);
+    }
+
+    Int32 size() const override
+    {
+        return 4;
+    }
+
+    MessageType getMessageType() const override
+    {
+        return MessageType::PARSE_COMPLETE;
+    }
+};
+
+class BindQuery : FrontMessage
+{
+public:
+    String portal_name;
+    String function_name;
+    std::vector<String> parameters;
+    Int32 num_params;
+
+    void deserialize(ReadBuffer & in) override
+    {
+        Int32 sz;
+        readBinaryBigEndian(sz, in);
+        readNullTerminated(portal_name, in);
+        readNullTerminated(function_name, in);
+
+        Int16 num_format_params;
+        readBinaryBigEndian(num_format_params, in);
+        Int16 format_param;
+        for (Int16 i = 0; i < num_format_params; ++i)
+            readBinaryBigEndian(format_param, in);
+
+        readBinaryBigEndian(num_params, in);
+        for (int i = 0; i < num_params; ++i)
+        {
+            Int32 sz_param;
+            readBinaryBigEndian(sz_param, in);
+            String current_param;
+            readNullTerminated(current_param, in);
+            parameters.push_back(current_param);
```

Comment:
this is not correct - according to docs it's not a null terminated string - we should read exact number of bytes in `sz_param`
https://www.postgresql.org/docs/current/protocol-message-formats.html#PROTOCOL-MESSAGE-FORMATS-BIND